arrow-fx-reactor/src/main/kotlin/arrow/fx/reactor/FluxK.kt

```diff
@@ -91,32 +93,45 @@ data class FluxK<out A>(val flux: Flux<out A>) : FluxKOf<A> {
    *  ```
    */
   fun <B> bracketCase(use: (A) -> FluxKOf<B>, release: (A, ExitCase<Throwable>) -> FluxKOf<Unit>): FluxK<B> =
-    FluxK(Flux.create<B> { sink ->
-      flux.subscribe({ a ->
-        if (sink.isCancelled) release(a, ExitCase.Cancelled).fix().flux.subscribe({}, sink::error)
-        else try {
-          sink.onDispose(use(a).fix()
-            .flatMap { b -> release(a, ExitCase.Completed).fix().map { b } }
-            .handleErrorWith { e -> release(a, ExitCase.Error(e)).fix().flatMap { FluxK.raiseError<B>(e) } }
-            .flux
-            .doOnCancel { release(a, ExitCase.Cancelled).fix().flux.subscribe({}, sink::error) }
-            .subscribe({ sink.next(it) }, sink::error, { }, {
-              sink.onRequest(it::request)
-            })
+    Flux.create<B> { emitter ->
+      val cancelable = ForwardDisposable()
+      emitter.onCancel(cancelable.cancel())
+
+      value().subscribe(
+        { a: A ->
+          cancelable.complete(
+            Flux.defer { use(a).value() }
+              .doOnCancel { Flux.defer { release(a, ExitCase.Cancelled).value() }.subscribe() }
+              .subscribe({ b ->
+                Flux.defer { release(a, ExitCase.Completed).value() }
+                  .subscribe({ emitter.next(b) }, { e -> emitter.error(e) })
+              }, { e ->
+                Flux.defer { release(a, ExitCase.Error(e)).value() }
+                  .subscribe({ emitter.error(e) }, { e2 -> emitter.error(Platform.composeErrors(e, e2)) })
+              }, {})
           )
-        } catch (e: Throwable) {
-          if (NonFatal(e)) {
-            release(a, ExitCase.Error(e)).fix().flux.subscribe({
-              sink.error(e)
-            }, { e2 ->
-              sink.error(Platform.composeErrors(e, e2))
-            })
-          } else {
-            throw e
-          }
-        }
-      }, sink::error, sink::complete)
-    })
+        },
+        { e: Throwable -> emitter.error(e) },
+        {}
```

Comment:
Same as mono, does completion need to be forwarded?

Reply:
The inner one needs to call `completion` after this one completes...

Needs fixing.